### ISSUE TITLE
Implement "Internet connection lost" & "Server down" banners (BannerGeneral) [$150 bounty]

### ISSUE THREAD

**Issue by taoeffect** [okTurtles/group-income#1060]
### Problem

At the moment we do not display the global `BannerGeneral` when an internet connection is lost, or the server is down.

Please see these links for full context:

- https://github.com/okTurtles/group-income-simple/issues/688#issuecomment-553473328
- https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?node-id=426%3A6

### Solution

Using the latest events implemented in @snowteamer's recent PR #1052, implement event handlers to display two different kinds of general banners (and hide them as appropriate):

1. When the Internet connection is down
2. When the Internet is up, but the server is down (e.g. via the ping/pong messages)

See `frontend/main.js` for examples of how to display the `BannerGeneral` component. E.g.:

```js
      sbp('okTurtles.data/set', 'BANNER', this.$refs.bannerGeneral)
      // call from anywhere in the app:
      // sbp('okTurtles.data/get', 'BANNER').show(L('Trying to reconnect...'), 'wifi')
      // sbp('okTurtles.data/get', 'BANNER').danger(L('message'), 'icon-type')
      // sbp('okTurtles.data/get', 'BANNER').clean()
```

### Bounty

$150 bounty for a clean solution to this (paid in cryptocurrency).

**Comment by snowteamer** [comment]
We can also use native `online` and `offline` events.

There are several possibilities as to where to implement the relevant event handlers:
- In the `handlers` option when the `GiPubSubClient` instance is created, using plain functions. That's probably the solution requiring the fewest lines of code.
- In the same file but just after the `createGiPubSubClient()` call, using more SBP calls.
- In `main.js` where the banner usage comments are currently located.
- In yet another file, maybe something like `general-banner-events.js`, if we do not want to clutter `main.js` with a bunch of event handlers.

**Comment by taoeffect** [comment]
@snowteamer if you'd like to go for this issue, then wherever you implement it, I recommend making sure all of the event handlers for these types of events (ping/pong stopped, online/offline) are grouped together in the same place. Fewer lines of code is better. So organized & simple is what we're aiming for. I agree that we shouldn't be cluttering up `main.js`. 👍

**Comment by snowteamer** [comment]
Another point: once an agreement has been reached regarding the messages to display, their French translation strings should be added in `french.strings`.

**Comment by snowteamer** [comment]
> At the moment we do not display the global BannerGeneral when an internet connection is lost, or the server is down.

@greg Please note that it may be technically difficult to distinguish between these two scenarios when all we know is that the server just does not respond to messages or does not send pings anymore.

The server might be down, but the connection might also have become broken without any `close` or `offline` event happening even if `navigator.onLine` is `true`, as that property being true is not a strong guarantee of having a working internet connection.

Still, it is not impossible, but for now I would suggest to opt for just displaying a message that would be equally suitable in both scenarios.

**Comment by taoeffect** [comment]
Closed in #1061
